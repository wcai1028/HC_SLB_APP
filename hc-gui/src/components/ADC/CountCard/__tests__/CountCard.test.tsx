import * as React from 'react'
import * as ReactTestRenderer from 'react-test-renderer'

const renderTestComponent = (props: IObject) => {
  const CountCard = require('../index').default

  return ReactTestRenderer.create(<CountCard {...props} />)
}

describe('should render CountCard', () => {
  it('should render with count', (done: jest.DoneCallback) => {
    const testComponent = renderTestComponent({
      title: 'Service',
      count: 10,
      color: '#8e8e8e',
      isWide: false,
    })
    setImmediate(() => {
      expect(testComponent.toJSON()).toMatchSnapshot()
      done()
    })
  })

  it('should render with counts', (done: jest.DoneCallback) => {
    const testComponent = renderTestComponent({
      title: 'Service',
      counts: [
        {
          color: '#fff',
          count: 10,
          name: 'HTTP',
        },
        {
          count: 10,
          name: 'TCP',
        },
      ],
      type: 'up',
      isWide: true,
    })
    setImmediate(() => {
      expect(testComponent.toJSON()).toMatchSnapshot()
      done()
    })
  })

  it('should render with up type', (done: jest.DoneCallback) => {
    const testComponent = renderTestComponent({
      title: 'Service',
      count: 10,
      type: 'up',
      isWide: false,
    })
    setImmediate(() => {
      expect(testComponent.toJSON()).toMatchSnapshot()
      done()
    })
  })

  it('should render with down type', (done: jest.DoneCallback) => {
    const testComponent = renderTestComponent({
      title: 'Service',
      count: 10,
      type: 'down',
      isWide: false,
    })
    setImmediate(() => {
      expect(testComponent.toJSON()).toMatchSnapshot()
      done()
    })
  })

  it('should render with partial type', (done: jest.DoneCallback) => {
    const testComponent = renderTestComponent({
      title: 'Service',
      count: 10,
      type: 'partial',
      isWide: false,
    })
    setImmediate(() => {
      expect(testComponent.toJSON()).toMatchSnapshot()
      done()
    })
  })

  it('should render with disable type', (done: jest.DoneCallback) => {
    const testComponent = renderTestComponent({
      title: 'Service',
      count: 10,
      type: 'disable',
      isWide: false,
    })
    setImmediate(() => {
      expect(testComponent.toJSON()).toMatchSnapshot()
      done()
    })
  })

  it('should render with unknow type', (done: jest.DoneCallback) => {
    const testComponent = renderTestComponent({
      title: 'Service',
      count: 10,
      type: 'unknow',
      isWide: false,
    })
    setImmediate(() => {
      expect(testComponent.toJSON()).toMatchSnapshot()
      done()
    })
  })
})
