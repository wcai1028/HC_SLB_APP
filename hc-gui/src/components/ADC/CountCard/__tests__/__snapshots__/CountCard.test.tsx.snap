// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CountCard should render with count 1`] = `
<div
  className="countCard"
  style={
    Object {
      "padding": "10px 10px 0 10px",
    }
  }
>
  <span
    className="countCardCount"
    style={
      Object {
        "color": "#8e8e8e",
      }
    }
  >
    10
  </span>
  <div
    className="countCardTitle"
    style={
      Object {
        "margin": "",
      }
    }
  >
    Service
  </div>
</div>
`;

exports[`should render CountCard should render with counts 1`] = `
<div
  className="countCard"
  style={
    Object {
      "padding": "0px 5px",
    }
  }
>
  <span
    className="countCardCount"
    style={
      Object {
        "color": "#7ed321",
      }
    }
  >
    <div
      className="countCardInner"
    >
      <div
        className="countCardInnerCount"
        style={
          Object {
            "color": "#fff",
          }
        }
      >
        10
      </div>
      <div
        className="countCardTitle"
        style={
          Object {
            "marginTop": "-6px",
          }
        }
      >
        HTTP
      </div>
    </div>
    <div
      className="countCardInner"
    >
      <div
        className="countCardInnerCount"
        style={
          Object {
            "color": "#8e8e8e",
          }
        }
      >
        10
      </div>
      <div
        className="countCardTitle"
        style={
          Object {
            "marginTop": "-6px",
          }
        }
      >
        TCP
      </div>
    </div>
  </span>
  <div
    className="countCardTitle"
    style={
      Object {
        "margin": "-4px 0 0 0",
      }
    }
  >
    Service
  </div>
</div>
`;

exports[`should render CountCard should render with disable type 1`] = `
<div
  className="countCard"
  style={
    Object {
      "padding": "10px 10px 0 10px",
    }
  }
>
  <span
    className="countCardCount"
    style={
      Object {
        "color": "#f1f8ff",
      }
    }
  >
    10
  </span>
  <div
    className="countCardTitle"
    style={
      Object {
        "margin": "",
      }
    }
  >
    Service
  </div>
</div>
`;

exports[`should render CountCard should render with down type 1`] = `
<div
  className="countCard"
  style={
    Object {
      "padding": "10px 10px 0 10px",
    }
  }
>
  <span
    className="countCardCount"
    style={
      Object {
        "color": "#ff2e48",
      }
    }
  >
    10
  </span>
  <div
    className="countCardTitle"
    style={
      Object {
        "margin": "",
      }
    }
  >
    Service
  </div>
</div>
`;

exports[`should render CountCard should render with partial type 1`] = `
<div
  className="countCard"
  style={
    Object {
      "padding": "10px 10px 0 10px",
    }
  }
>
  <span
    className="countCardCount"
    style={
      Object {
        "color": "#f5a623",
      }
    }
  >
    10
  </span>
  <div
    className="countCardTitle"
    style={
      Object {
        "margin": "",
      }
    }
  >
    Service
  </div>
</div>
`;

exports[`should render CountCard should render with unknow type 1`] = `
<div
  className="countCard"
  style={
    Object {
      "padding": "10px 10px 0 10px",
    }
  }
>
  <span
    className="countCardCount"
    style={
      Object {
        "color": "#ffba47",
      }
    }
  >
    10
  </span>
  <div
    className="countCardTitle"
    style={
      Object {
        "margin": "",
      }
    }
  >
    Service
  </div>
</div>
`;

exports[`should render CountCard should render with up type 1`] = `
<div
  className="countCard"
  style={
    Object {
      "padding": "10px 10px 0 10px",
    }
  }
>
  <span
    className="countCardCount"
    style={
      Object {
        "color": "#7ed321",
      }
    }
  >
    10
  </span>
  <div
    className="countCardTitle"
    style={
      Object {
        "margin": "",
      }
    }
  >
    Service
  </div>
</div>
`;
